src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs

```diff
@@ -10,16 +10,35 @@
 
 namespace System.Text.Encodings.Web
 {
+    /// <summary>
+    /// Represents a type used to do HTML encoding.
+    /// </summary>
     public abstract class HtmlEncoder : TextEncoder
     {
+        /// <summary>
+        /// Returns default built-in instance of HtmlEncoder.
+        /// </summary>
         public static HtmlEncoder Default
         {
             get { return DefaultHtmlEncoder.Singleton; }
         }
+
+        /// <summary>
+        /// Creates a new instalce of HtmlEncoder with provided settings.
+        /// </summary>
+        /// <param name="settings">Settings used mainly to specify which characters end up being encoded.</param>
+        /// <returns>Returns new instance of the encoder.</returns>
         public static HtmlEncoder Create(TextEncoderSettings settings)
         {
             return new DefaultHtmlEncoder(settings);
         }
+
+        /// <summary>
+        /// Creates a new instalce of HtmlEncoder specifying character to be encoded.
+        /// </summary>
+        /// <param name="allowedRanges">Set of characters that the endocer is allowed to not encode.</param>
+        /// <returns>Returns new instance of the encoder.</returns>
+        /// <remarks>Some characters in the ranges might still get encoded, i.e. this parameter is just a hint.</remarks> 
```

Comment:
This type is an abstraction, so it's not clear to me if examples would be useful. The point is that the allowedRanges is just telling the encoder what ranges it is allowed to not encode; it is not ordering it to not encode them. Implementations may still hardcode some special characters and to always encode them.

Maybe I just say: 
"Some characters in the ranges might still get encoded, i.e. this parameter is just telling the encoder what ranges it is allowed to not encode, not what characters it must not encode."